software/soapysdr-xtrx/Streaming.cpp

```diff
@@ -312,25 +313,36 @@ int SoapyXTRX::acquireWriteBuffer(SoapySDR::Stream *stream, size_t &handle,
         assert(buffers_pending < ((int64_t)_dma_mmap_info.dma_tx_buf_count));
     }
 
-    // get the buffer
-    int buf_offset = _tx_stream.user_count % _dma_mmap_info.dma_tx_buf_count;
-    getDirectAccessBufferAddrs(stream, buf_offset, buffs);
-
-    // update the DMA counters
-    handle = _tx_stream.user_count;
-    _tx_stream.user_count++;
-
-    // detect underflows
+    // detect underflows of the underlying circular buffer
     if (buffers_pending < 0) {
+        // drain all buffers to get out of the underflow quicker
+        struct litepcie_ioctl_mmap_dma_update mmap_dma_update;
+        mmap_dma_update.sw_count = _tx_stream.hw_count;
+        checked_ioctl(_fd, LITEPCIE_IOCTL_MMAP_DMA_READER_UPDATE, &mmap_dma_update);
+        _tx_stream.user_count = _tx_stream.hw_count;
+        _tx_stream.sw_count = _tx_stream.hw_count;
+        handle = -1;
+
```

Comment:
So this basically says "couldn't transmit, and I'm zooming backward so that we have 255 buffers of space between the read pointer and the write pointer", right?  If that's the case, why can't we write into a buffer here?  Is it just so that we can surface errors better?

Reply:
This logic needs to be revisited. I have removed it for now. We can add it back later.